tests/subsys/fs/nvs/src/main.c

```diff
@@ -959,3 +959,40 @@ ZTEST_F(nvs, test_nvs_cache_hash_quality)
 
 #endif
 }
+
+/*
+ * Test NVS bad region initialization recovery.
+ */
+ZTEST_F(nvs, test_nvs_init_bad_memory_region)
+{
+	int err;
+	uint32_t data;
+
+	err = nvs_mount(&fixture->fs);
+	zassert_true(err == 0, "nvs_mount call failure: %d", err);
+
+	/* Write bad ATE to each sector */
+	for (uint16_t i = 0; i < TEST_SECTOR_COUNT; i++) {
+		data = 0xdeadbeef;
+		err = flash_write(fixture->fs.flash_device,
+			fixture->fs.offset +
```

Comment:
The alignment should be with the start of `fixture->fs.flash_device`. The complete file however is no example for correct alignment :-).